src/engines/html5/html5.js

```diff
@@ -194,13 +212,85 @@ export default class Html5 extends FakeEventTarget implements IEngine {
 
   /**
    * Select a new text track.
-   * @param {TextTrack} textTrack - The text track object to set.
+   * @param {PKTextTrack} textTrack - The text track object to set.
    * @returns {void}
    */
-  selectTextTrack(textTrack: TextTrack): void {
+  selectTextTrack(textTrack: PKTextTrack): void {
+    this._removeCueChangeListener();
     if (this._mediaSourceAdapter) {
       this._mediaSourceAdapter.selectTextTrack(textTrack);
     }
+    this._addCueChangeListener(textTrack);
+  }
+
+  /**
+   * Add cuechange listener to active textTrack
+   * @param {PKTextTrack} textTrack - player text track
+   * @returns {void}
+   * @private
+   */
+  _addCueChangeListener(textTrack: PKTextTrack): void {
+    let textTrackEl = this._getSelectedTextTrack();
+    if (textTrackEl) {
+      //there's a quirk in TextTrackAPI that a text track added to video element will not fire cuechange event if it
+      //didn't have it's mode set to showing for at least until a single cue has been change.
+      //After first time it seems there's time tracking which allows the cuechange to fire even though the track mode
+      //is set to hidden
+      //This is not the case with a track DOM element added to a video element where cuechange will be fired even if
+      // track mode is set only to hidden and was never set to showing
+      if (this._config.playback.useNativeTextTrack) {
+        textTrackEl.mode = "showing";
+      } else {
+        textTrackEl.mode = this._showTextTrackfirstTime[textTrack.index] ? "hidden" : "showing";
+        this._showTextTrackfirstTime[textTrack.index] = true;
+        textTrackEl.oncuechange = (e) => this._onCueChange(e);
+      }
+    }
+  }
+
+  /**
+   * Remove cuechange listener to active textTrack
+   * @param {PKTextTrack} textTrack - player text track
```

Comment:
**playkit** text track

Reply:
ooopps - I don't even use it so no ned to define it